--- a/Source/WebCore/platform/graphics/transforms/TransformationMatrix.cpp
+++ b/Source/WebCore/platform/graphics/transforms/TransformationMatrix.cpp
@@ -1241,7 +1241,7 @@ TransformationMatrix& TransformationMatrix::multiply(const TransformationMatrix&
     }
 #undef MATRIX_MULTIPLY_ONE_LINE
 
-#elif defined(TRANSFORMATION_MATRIX_USE_X86_64_SSE2)
+#elif 0
     // x86_64 has 16 XMM registers which is enough to do the multiplication fully in registers.
     __m128d matrixBlockA = _mm_load_pd(&(m_matrix[0][0]));
     __m128d matrixBlockC = _mm_load_pd(&(m_matrix[1][0]));
