diff -up webkitgtk-2.13.90/Source/WebKit2/UIProcess/API/gtk/WebKitSettings.cpp.disable_npapi_in_wayland webkitgtk-2.13.90/Source/WebKit2/UIProcess/API/gtk/WebKitSettings.cpp
--- webkitgtk-2.13.90/Source/WebKit2/UIProcess/API/gtk/WebKitSettings.cpp.disable_npapi_in_wayland	2016-08-30 18:52:41.000000000 +0200
+++ webkitgtk-2.13.90/Source/WebKit2/UIProcess/API/gtk/WebKitSettings.cpp	2016-08-31 12:59:03.398097549 +0200
@@ -1621,6 +1621,12 @@ void webkit_settings_set_enable_plugins(
 {
     g_return_if_fail(WEBKIT_IS_SETTINGS(settings));
 
+#if PLATFORM(WAYLAND)
+    // https://bugs.webkit.org/show_bug.cgi?id=158697
+    if (WebCore::PlatformDisplay::sharedDisplay().type() == WebCore::PlatformDisplay::Type::Wayland)
+        return;
+#endif
+
     WebKitSettingsPrivate* priv = settings->priv;
     bool currentValue = priv->preferences->pluginsEnabled();
     if (currentValue == enabled)
diff -up webkitgtk-2.13.90/Source/WebKit2/UIProcess/gtk/WebPreferencesGtk.cpp.disable_npapi_in_wayland webkitgtk-2.13.90/Source/WebKit2/UIProcess/gtk/WebPreferencesGtk.cpp
--- webkitgtk-2.13.90/Source/WebKit2/UIProcess/gtk/WebPreferencesGtk.cpp.disable_npapi_in_wayland	2016-08-30 18:52:41.000000000 +0200
+++ webkitgtk-2.13.90/Source/WebKit2/UIProcess/gtk/WebPreferencesGtk.cpp	2016-08-31 12:52:10.954610986 +0200
@@ -69,6 +69,9 @@ void WebPreferences::platformInitializeS
     if (PlatformDisplay::sharedDisplay().type() == PlatformDisplay::Type::Wayland) {
         if (!WaylandCompositor::singleton().isRunning())
             setAcceleratedCompositingEnabled(false);
+
+        // https://bugs.webkit.org/show_bug.cgi?id=158697
+        setPluginsEnabled(false);
     }
 #endif
 
