diff -up webkitgtk-2.5.90/Source/JavaScriptCore/llint/LowLevelInterpreter.asm.cloop_fix_32 webkitgtk-2.5.90/Source/JavaScriptCore/llint/LowLevelInterpreter.asm
--- webkitgtk-2.5.90/Source/JavaScriptCore/llint/LowLevelInterpreter.asm.cloop_fix_32	2014-09-22 13:53:39.777953541 +0200
+++ webkitgtk-2.5.90/Source/JavaScriptCore/llint/LowLevelInterpreter.asm	2014-09-22 13:54:32.813718555 +0200
@@ -533,13 +533,13 @@ macro assertNotConstant(index)
 end
 
 macro functionForCallCodeBlockGetter(targetRegister)
-    loadp Callee[cfr], targetRegister
+    loadp Callee + PayloadOffset[cfr], targetRegister
     loadp JSFunction::m_executable[targetRegister], targetRegister
     loadp FunctionExecutable::m_codeBlockForCall[targetRegister], targetRegister
 end
 
 macro functionForConstructCodeBlockGetter(targetRegister)
-    loadp Callee[cfr], targetRegister
+    loadp Callee + PayloadOffset[cfr], targetRegister
     loadp JSFunction::m_executable[targetRegister], targetRegister
     loadp FunctionExecutable::m_codeBlockForConstruct[targetRegister], targetRegister
 end
