diff -up webkitgtk-2.10.4/Source/WebKit2/Shared/Plugins/unix/PluginSearchPath.cpp.nspluginwrapper webkitgtk-2.10.4/Source/WebKit2/Shared/Plugins/unix/PluginSearchPath.cpp
--- webkitgtk-2.10.4/Source/WebKit2/Shared/Plugins/unix/PluginSearchPath.cpp.nspluginwrapper	2015-11-11 11:46:43.848699104 +0100
+++ webkitgtk-2.10.4/Source/WebKit2/Shared/Plugins/unix/PluginSearchPath.cpp	2015-11-11 11:47:44.450490935 +0100
@@ -68,6 +68,9 @@ Vector<String> pluginsDirectories()
     result.append("/usr/lib64/mozilla/plugins");
     result.append("/usr/lib/nsbrowser/plugins");
     result.append("/usr/lib64/nsbrowser/plugins");
+    /* nspluginwrapper plugins */
+    result.append("/usr/lib64/mozilla/plugins-wrapped");
+    result.append("/usr/lib/mozilla/plugins-wrapped");
 #endif
 
     return result;
