diff -up webkitgtk-2.7.3/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp.disable_codec_installer webkitgtk-2.7.3/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp
--- webkitgtk-2.7.3/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp.disable_codec_installer	2014-12-16 12:51:22.476697486 +0100
+++ webkitgtk-2.7.3/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp	2014-12-16 12:53:13.921359868 +0100
@@ -1020,15 +1020,8 @@ gboolean MediaPlayerPrivateGStreamer::ha
         gst_bin_recalculate_latency(GST_BIN(m_playBin.get()));
         break;
     case GST_MESSAGE_ELEMENT:
-        if (gst_is_missing_plugin_message(message)) {
-            gchar* detail = gst_missing_plugin_message_get_installer_detail(message);
-            gchar* detailArray[2] = {detail, 0};
-            GstInstallPluginsReturn result = gst_install_plugins_async(detailArray, 0, mediaPlayerPrivatePluginInstallerResultFunction, this);
-            m_missingPlugins = result == GST_INSTALL_PLUGINS_STARTED_OK;
-            g_free(detail);
-        }
 #if ENABLE(VIDEO_TRACK) && USE(GSTREAMER_MPEGTS)
-        else {
+        if (!gst_is_missing_plugin_message(message)) {
             GstMpegtsSection* section = gst_message_parse_mpegts_section(message);
             if (section) {
                 processMpegTsSection(section);
