diff -up webkitgtk-2.4.2/Source/WTF/wtf/dtoa/utils.h.ppc64le webkitgtk-2.4.2/Source/WTF/wtf/dtoa/utils.h
--- webkitgtk-2.4.2/Source/WTF/wtf/dtoa/utils.h.ppc64le	2014-05-14 15:37:26.360159487 +0200
+++ webkitgtk-2.4.2/Source/WTF/wtf/dtoa/utils.h	2014-05-14 15:38:46.938315499 +0200
@@ -49,7 +49,7 @@
 defined(__ARMEL__) || \
 defined(_MIPS_ARCH_MIPS32R2)
 #define DOUBLE_CONVERSION_CORRECT_DOUBLE_OPERATIONS 1
-#elif CPU(MIPS) || CPU(PPC) || CPU(PPC64) || OS(WINCE) || CPU(SH4) || CPU(S390) || CPU(S390X) || CPU(IA64) || CPU(ALPHA) || CPU(ARM64) || CPU(HPPA)
+#elif CPU(MIPS) || CPU(PPC) || CPU(PPC64) || CPU(PPC64LE) || OS(WINCE) || CPU(SH4) || CPU(S390) || CPU(S390X) || CPU(IA64) || CPU(ALPHA) || CPU(ARM64) || CPU(HPPA)
 #define DOUBLE_CONVERSION_CORRECT_DOUBLE_OPERATIONS 1
 #elif defined(_M_IX86) || defined(__i386__)
 #if defined(_WIN32)
diff -up webkitgtk-2.4.2/Source/WTF/wtf/Platform.h.ppc64le webkitgtk-2.4.2/Source/WTF/wtf/Platform.h
--- webkitgtk-2.4.2/Source/WTF/wtf/Platform.h.ppc64le	2014-05-12 08:03:53.000000000 +0200
+++ webkitgtk-2.4.2/Source/WTF/wtf/Platform.h	2014-05-14 15:39:08.402622454 +0200
@@ -101,24 +101,38 @@
 #endif /* MIPS */
 
 /* CPU(PPC) - PowerPC 32-bit */
-#if   defined(__ppc__)     \
+#if (  defined(__ppc__)     \
     || defined(__PPC__)     \
     || defined(__powerpc__) \
     || defined(__powerpc)   \
     || defined(__POWERPC__) \
     || defined(_M_PPC)      \
-    || defined(__PPC)
+    || defined(__PPC)) \
+    && defined(__BYTE_ORDER__) \
+    && (__BYTE_ORDER__==__ORDER_BIG_ENDIAN__)
 #define WTF_CPU_PPC 1
 #define WTF_CPU_BIG_ENDIAN 1
 #endif
 
-/* CPU(PPC64) - PowerPC 64-bit */
-#if   defined(__ppc64__) \
-    || defined(__PPC64__)
+/* CPU(PPC64) - PowerPC 64-bit Big Endian */
+#if (  defined(__ppc64__) \
+    || defined(__PPC64__)) \
+    && defined(__BYTE_ORDER__) \
+    && (__BYTE_ORDER__==__ORDER_BIG_ENDIAN__)
 #define WTF_CPU_PPC64 1
 #define WTF_CPU_BIG_ENDIAN 1
 #endif
 
+/* CPU(PPC64) - PowerPC 64-bit Little Endian */
+#if (   defined(__ppc64__) \
+    || defined(__PPC64__) \
+    || defined(__ppc64le__) \
+    || defined(__PPC64LE__)) \
+    && defined(__BYTE_ORDER__) \
+    && (__BYTE_ORDER__==__ORDER_LITTLE_ENDIAN__)
+#define WTF_CPU_PPC64LE 1
+#endif
+
 /* CPU(SH4) - SuperH SH-4 */
 #if defined(__SH4__)
 #define WTF_CPU_SH4 1
@@ -641,7 +655,8 @@
     || CPU(ALPHA) \
     || CPU(ARM64) \
     || CPU(S390X) \
-    || CPU(PPC64)
+    || CPU(PPC64) \
+    || CPU(PPC64LE)
 #define WTF_USE_JSVALUE64 1
 #else
 #define WTF_USE_JSVALUE32_64 1
