diff -up webkitgtk-2.7.90/Source/JavaScriptCore/llint/LLIntSlowPaths.cpp.180248 webkitgtk-2.7.90/Source/JavaScriptCore/llint/LLIntSlowPaths.cpp
--- webkitgtk-2.7.90/Source/JavaScriptCore/llint/LLIntSlowPaths.cpp.180248	2015-02-17 08:08:23.000000000 +0100
+++ webkitgtk-2.7.90/Source/JavaScriptCore/llint/LLIntSlowPaths.cpp	2015-02-18 11:15:39.060933863 +0100
@@ -490,6 +490,7 @@ LLINT_SLOW_PATH_DECL(stack_check)
         LLINT_RETURN_TWO(pc, 0);
 #endif
 
+    exec = exec->callerFrame(vm.topVMEntryFrame);
     vm.topCallFrame = exec;
     ErrorHandlingScope errorScope(vm);
     CommonSlowPaths::interpreterThrowInCaller(exec, createStackOverflowError(exec));
